web/modules/custom/bnf/src/Plugin/bnf_mapper/ParagraphGoLinkboxMapper.php

```diff
@@ -47,33 +48,28 @@ public function map(ObjectLike $object): mixed {
       throw new \RuntimeException('Wrong class handed to mapper');
     }
 
-    /** @var \Drupal\bnf\GraphQL\Operations\GetNode\Node\Paragraphs\GoLinkParagraph\ParagraphGoLink $goLink */
-    $goLink = $object->goLinkParagraph;
-
-    $goLinkParagraph = $this->paragraphStorage->create([
-      'type' => 'go_link',
-      'field_aria_label' => $goLink->ariaLabel,
-      'field_target_blank' => $goLink->targetBlank,
-      'field_go_link' => $this->getLinkValue($goLink->linkRequired),
-    ]);
-
-    /** @var \Drupal\paragraphs\Entity\Paragraph $goLinkParagraph */
+    /** @var null|\Drupal\paragraphs\Entity\Paragraph $goLinkParagraph */
+    $goLinkParagraph = $this->mapper->map($object->goLinkParagraph);
 
-    $goLinkParagraph->save();
-
-    return $this->paragraphStorage->create([
+    /** @var \Drupal\paragraphs\Entity\Paragraph $linkbox */
+    $linkbox = $this->paragraphStorage->create([
       'type' => 'go_linkbox',
       'field_go_color' => $object->goColor,
       'field_go_description' => $object->goDescription,
       'field_go_image' => $this->getImageValue($object->goImage),
-      'field_go_link_paragraph' => [[
-        'target_id' => $goLinkParagraph->id(),
-        'target_revision_id' => $goLinkParagraph->getRevisionId(),
-      ],
+      'field_go_link_paragraph' => [
+        $goLinkParagraph,
```

Comment:
?? Below you're doing the same, although in a `if ($goLinkParagraph) {`?

Reply:
The go link mapper can return `null` if we hit the recursion limit, so we should only set the field when we have a value.

Reply:
i don't understand your answer here.

if you look at your code, what you're doing at line 60-62 is identical to what you're doing at line 66-72 - but at 66-72, you're actually checking if the value exists